### PR TITLE
Fix up CI for deprecation alerts.

### PR DESCRIPTION
**Pull request by dereuromark**
Reopen https://github.com/cakephp/cakephp/pull/17357

Also, now visible with the script running again:
```
Cake\ORM\Query\SelectQuery
 * Missing class_exists() or class_alias() on new file for ORM/Query.php => ORM/Query/SelectQuery.php
```

Seems like there is one missing from the other direction, there is an alias on the SelectQuery, but not on the Query itself.

### CODE REVIEW

**Comment by othercorey** [comment]
I think the problem is in Query.php. It should have class_exists not class_alias.

**Comment by dereuromark** [comment]
Jep, you are right, should now be properly fixed up
```
Cake\I18n\Date
 * OK
Cake\I18n\DateTime
 * OK
Cake\Controller\ControllerFactory
 * OK
Cake\Database\TypeFactory
 * OK
Cake\ORM\Query\SelectQuery
 * OK
```

File: src/ORM/Query/SelectQuery.php [hunk 1]
```diff
@@ -1795,3 +1795,7 @@ public function isAutoFieldsEnabled(): ?bool
         return $this->_autoFields;
     }
 }
+
+// phpcs:disable
+class_exists('Cake\ORM\Query');
```
Comment (ADmad) [inline src/ORM/Query/SelectQuery.php:1800]: I thought we would have dropped ORM\Query in 5.0, it's been deprecated since 4.4.

Comment (ADmad) [reply src/ORM/Query/SelectQuery.php:1800]: Err my bad, it's deprecated since 4.5 not 4.4.

**Pull request merged**